File: Palindrome String - GFG/palindrome-string.cpp
class Solution{
public:	
	int isPalindrome(string s){
	    // Your code goes here
	    int n = s.size();
	    for(int i=0; i<n/2; i++){
	        if(s[i] != s[n-i-1]){
	            return 0;
	        }
	    }
	    return 1;
	}
};

File: Palindrome String - GFG/palindrome-string.py
#User function Template for python3
class Solution:
	def isPalindrome(self, S):
		# code here
		pali = S[::-1]
		if pali==S:
		    return 1
		return 0


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
	T=int(input())
	for i in range(T):
		S = input()
		ob = Solution()
		answer = ob.isPalindrome(S)
		print(answer)

# } Driver Code Ends